waiter/src/waiter/token.clj

```diff
@@ -157,6 +169,25 @@
                   token-set (->> tokens (map :token) set)]
               (kv/store kv-store owner-key token-set))))))))
 
+;; A token store which is backed by a key-value store.
+(defrecord KeyValueTokenStore [clock synchronize-fn kv-store]
+
+  TokenStore
+
+  (fetch [_ token include-deleted?]
+    (sd/token->token-description kv-store token :include-deleted include-deleted?))
+
+  (store [_ token service-description-template token-metadata]
+    ;; insert the last-update-time if it doesn't exist
+    (let [token-metadata' (merge {"last-update-time" (.getMillis ^DateTime (clock))} token-metadata)]
+      (store-service-description-for-token synchronize-fn kv-store token service-description-template token-metadata')))
```

Comment:
Inline `store-service-description-for-token`

Reply:
Inlining requires moving the procol inside the `let` block that defines `token-lock`. Protocols don't appear to recognize variables defined in let blocks, I get the following error:
`Unable to resolve symbol: token-lock in this context, compiling:(waiter/token.clj:138:9)`.

I wrote this simple example and had the same error:
```
(defprotocol P (foo [this]))
(let [x 1]
  (defrecord R [] P (foo [_] (println x))))
=> 
CompilerException java.lang.RuntimeException: Unable to resolve symbol: x in this context, compiling:(/private/var/folders/wj/c9w0p0r93vl4rl815rspwxqw0000gn/T/form-init7112261126071510316.clj:3:30) 
```